ingestion/tests/magen_ingestion_test.py

```diff
@@ -1138,6 +1186,44 @@ def test_create_meta(self):
         ret = EncryptionApi.create_meta("c9243e28-238d-41b6-9c5e-37f0b1be4dae")
         self.assertEqual(orig, ret)
 
+    def test_JQuery_UploadFile_with_Public_Key(self):
+        """
+        This test simulates the jquery-file-upload client. It uploads a file through POST form data.
+        It checks the response was 200OK.
+        """
+        print("+++++++++ test_JQuery_UploadFile_with_Public_Key Test +++++++++")
+
+        server_urls_instance = ServerUrls().get_instance()
+        file_name = "test_up.pub"
+        base_path = type(self).ingestion_globals.data_dir
+        file_full_path = os.path.join(base_path, file_name)
+        delete_url = None
+        try:
+            magen_file = open(file_full_path, 'w+')
+            magen_file.write("this is a test")
+            magen_file.close()
+            files = {'files[]': (file_full_path, file_name, 'text/plain')}
+            jquery_file_upload_url = server_urls_instance.ingestion_server_base_url + "file_upload/"
+            post_resp_obj = type(self).app.post(jquery_file_upload_url, data=files,
+                                                headers={'content-type': 'multipart/form-data'})
+            self.assertEqual(post_resp_obj.status_code, HTTPStatus.OK)
+            post_resp_json_obj = json.loads(post_resp_obj.data.decode("utf-8"))
+            delete_url = post_resp_json_obj["files"][0]["url"]
+
```

Comment:
As we discussed, you need to retrieve the file from GridFS to make sure it matches the one you uploaded.